### PR TITLE
Fix/miner

### PR DESCRIPTION
**Pull request by ToMe25**
# Changes proposed in this pull request:
- Make the builder repair its ladder instead of just filling everything below a missing one with cobble and starting from there.
- Make the mineshaft nodes be always on the same heights(every 10 blocks, not where they are designed to be, as i didn't know they were designed to be only one per builder level.).

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java [hunk 1]
```diff
@@ -398,17 +445,45 @@ private IAIState advanceLadder(final IAIState state)
             return state;
         }
 
+        @NotNull final BlockPos nextLadder =
+                new BlockPos(getOwnBuilding().getLadderLocation().getX(), getLastLadder(getOwnBuilding().getLadderLocation(), world) - 1, getOwnBuilding().getLadderLocation().getZ());
+        final int xOffset = SHAFT_RADIUS * getOwnBuilding().getVectorX();
+        final int zOffset = SHAFT_RADIUS * getOwnBuilding().getVectorZ();
+        boolean repair = true;
+
+        if (world.getBlockState(nextLadder).getBlock() instanceof AirBlock)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java]: might make sense to put this in a utility functin that returns true if repair and else false

Comment (ToMe25) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java]: yes, but most of this stuff will get removed in the next commit.

Comment (ToMe25) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java]: as it checked the wrong stuff

Comment (ToMe25) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java]: This should be solved now

Comment (ToMe25) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java]: ok now i think i understand what you meant.
1. i thought this was the other check that looks the same
2. yes, i will probably do that.

Comment (ToMe25) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java]: ok i did this.

**Comment by ToMe25** [comment]
I think this PR is ready now.

**Review by marchermans — APPROVED** [review]

**Review by marchermans — APPROVED** [review]

**Review by Asherslab — APPROVED** [review]

**Pull request merged**